shell/browser/api/electron_api_utility_process.cc

```diff
@@ -408,25 +202,41 @@ void UtilityProcessWrapper::OnServiceProcessLaunched(
   DCHECK(node_service_remote_.is_connected());
   pid_ = process.Pid();
   GetAllUtilityProcessWrappers().AddWithID(this, pid_);
+  if (stdout_read_fd_ != -1) {
+    EmitWithoutCustomEvent("stdout", stdout_read_fd_);
+  }
+  if (stderr_read_fd_ != -1) {
+    EmitWithoutCustomEvent("stderr", stderr_read_fd_);
+  }
   // Emit 'spawn' event
   EmitWithoutCustomEvent("spawn");
 }
 
 void UtilityProcessWrapper::OnServiceProcessDisconnected(
     uint32_t error_code,
     const std::string& description) {
-  Shutdown(0 /* exit_code */);
+  if (pid_ != base::kNullProcessId)
+    GetAllUtilityProcessWrappers().Remove(pid_);
+  CloseConnectorPort();
+  // Emit 'exit' event
+  EmitWithoutCustomEvent("exit", error_code);
+  Unpin();
+}
+
+void UtilityProcessWrapper::CloseConnectorPort() {
+  if (!connector_closed_ && connector_->is_valid()) {
```

Comment:
Does this need a separate boolean to track whether the connector has been closed? is `connector_ != nullptr` always identical?

Reply:
I don't think we need another boolean, `connector_` is never `nullptr` unless `UtilityProcessWrapper::CloseConnectorPort` is called and it only gets called once when the the process exits.